src/current-user/CurrentUser.js

```diff
@@ -157,15 +157,17 @@ class CurrentUser {
      *
      * The request done is equivalent do doing https://play.dhis2.org/demo/api/27/me.json?fields=userGroups[:all]
      *
+     * @param {Object} [listOptions={}] Additional query parameters that should be send with the request.
      * @returns {Promise<ModelCollection>} The model collection that contains the user's groups.
      */
-    getUserGroups() {
+    getUserGroups(listOptions = {}) {
         const userGroupIds = this[propertySymbols.userGroups]
 
-        return this[models].userGroup.list({
-            filter: [`id:in:[${userGroupIds.join(',')}]`],
-            paging: false,
-        })
+        return this[models].userGroup.list(
+            Object.assign({ paging: false }, listOptions, {
+                filter: [`id:in:[${userGroupIds.join(',')}]`],
+            })
```

Comment:
If the intention is to allow overwriting the `filter` property, then the `listOptions` should be the last argument in the `assign`.

Reply:
True, I used the same approach already in place for `getOrganisationUnits`.
I don't think the `filter` should be overridden, these methods are called like `d2.currentUser.getUserGroups()` which implies the groups returned are the ones where the current user belongs.

Reply:
I think you make a good point, but we probably need a slightly more nuanced solution:
```js
{
    paging: false,
    ...listOptions,
    filter: [
        ...listOptions.filter,
        `id:in:[${userGroupIds.join(',')}]`,
     ]
 }   
```
My point: the part where we filter the userGroups on the user's userGroups should not be overwritten.